api/src/lib.rs

```diff
@@ -44,16 +45,19 @@ pub struct Environment {
     pub variables: Vec<(String, String)>,
 }
 
+#[derive(Clone)]
 pub struct PostieApi {
-    pub environment: Option<String>,
+    pub client: reqwest::Client,
```

Comment:
The thought here is making it easier to test if we can inject a mock client

Reply:
+1 for dependency injection in compiled languages